### ISSUE TITLE
TooltipHost's tooltip text can remain in DOM, causing Narrator in scan mode to incorrectly focus on & read it

### ISSUE THREAD

**Issue by jendowns** [microsoft/fluentui#16308]
### Environment Information

- **Package version(s)**:  `"office-ui-fabric-react": "^7.135.0",`
- **Browser and OS versions**: Windows 10 Enterprise (build 19042.685), Edge 87.0.664.66

### Describe the issue:

If a `TooltipHost`'s tooltip is activated with a mouse, its text node can remain in the DOM even after that tooltip appears visually dismissed. This can cause problems for Narrator in scan mode, which will attempt to focus on & read that text node. For example, if you have an `IconButton` wrapped in a `TooltipHost` & the tooltip text remains in the DOM, Narrator in scan mode will focus on & read out that button's info twice -- once for the `IconButton`'s assistive text and once for its tooltip text that is still in the DOM.

You can reproduce this bug on the FluentUI doc site directly (steps & video below). This bug was originally filed in our ADO repository on a control that has multiple `IconButton`s (each wrapped in `TooltipHost`) side by side. It's possible there are other situations where this bug happens with the `TooltipHost`, the lingering tooltip text, and Narrator; it's just that it's especially noticeable when you have an `IconButton` because it already has its own assistive text & this bug causes Narrator to have the double focus + read-out for a single button.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/jendowns/pen/gOwGwPq

1. Turn on Narrator (scan mode)
2. Click on the first `IconButton` with the tooltip. Note that the tooltip is shown while your mouse is hovering over the button and also that Narrator reads out the button's assistive text.
3. Click on the down arrow key to use scan mode to move to the next control for Narrator to read. Note that Narrator reads the original button's tooltip text (which is no longer visible) while focusing on a small 0px by 0px space in the corner of the button.
4. Use arrows keys to go back and forth between the `IconButton`s in the example, and note that one button keeps being focused & read twice -- once for its assistive text, and once for tooltip text that is still in the DOM. You can inspect the page to see the tooltip text is still there, and you can see it is not ignored in the accessibility tree, which causes Narrator to read it.

#### Actual behavior:

The tooltip for a `TooltipHost` wrapping an `IconButton` is sometimes not removed from the DOM when you alternate between clicking and using arrow keys in Narrator scan mode.  As a result, using scan mode to travel back and forth can cause it to focus on and read out the `IconButton`'s info twice -- once for its assistive text, and once for its tooltip text.

#### Expected behavior:

The tooltip for the `TooltipHost` wrapping an `IconButton` should be removed from the DOM when the button is no longer active / focused on. Narrator should not focus on visually hidden, non-assistive text and read it out.

https://user-images.githubusercontent.com/9057921/103257026-24009d80-4955-11eb-9c55-f4707ec07252.mp4

**Comment by brandonthomas** [comment]
@ecraig12345 @paulgildea ping on this.

**Comment by gouttierre** [comment]
Thanks for the feedback and providing your findings.. 

@khmakoto  - Would you be able to determine if this is a regression or behavior where the DOM is occasionally removed when switching between clicking and using arrow key functions? cc @smhigley

**Comment by khmakoto** [comment]
@jendowns @brandonthomas I think this is an issue of Narrator's scan mode not playing well with the 250ms delay we have in our `Tooltips` by default, moving into the `Tooltip` text before it dismisses and not dismissing it afterwards since it is "focused". If the delay is set to 0 this doesn't happen as can be seen in [this codepen](https://codepen.io/khmakoto/pen/dyOwLQL?editors=0010).

Given all of this, I think I'm going to resolve this as "External" since it's not an issue with our code itself but on how Narrator's scan mode works. Let me know if there is anything else I can do to help!

**Comment by brandonthomas** [comment]
@jendowns let's follow up with Mohammad on the a11y team then and get this logged against Narrator so they can take a look.

Thanks @khmakoto for taking a look.

**Comment by jendowns** [comment]
Hey @khmakoto, thank you for investigating!

However, I don't believe this is a Narrator bug; I believe it's a tooltip-related bug. It's just that the bug is more noticeable (and causes problems) with Narrator, which reads the visually hidden text. The bug can be demonstrated without using a screen reader at all.

**I think the CodePen example you made, where you set the delay to 0, may be a good clue for the cause of the bug. In my CodePen example with the normal delay (https://codepen.io/jendowns/pen/gOwGwPq), I'm able to demonstrate the bug without Narrator with 3 actions:**

1.  left-click on the first button
2.  tab to next button
3.  left-click elsewhere on page to remove focus

_(NOTE: in the last few frames of the video, I hover over the tooltip element with the text "Emoji 1" that is left in the DOM.)_

https://user-images.githubusercontent.com/9057921/112365092-3dd82500-8ca5-11eb-93c4-df9d4b01b369.mp4

This to me indicates there's an issue with the tooltip dismiss events + the delay. And to be clear, I can't replicate the bug in the same way in your CodePen example (with delay set to 0), which is a good clue I think.

I really think this bug should be reopened and investigated, specifically looking at the way the tooltip text is dismissed + removed from the DOM and considering how the built-in delay plays into that. Please let me know what you think -- thank you!

cc @brandonthomas

**Comment by khmakoto** [comment]
@jendowns I'm reopening this issue. Thanks for the explanation, I agree that this needs to be investigated further.
